### PR TITLE
(#1237) Upgrade to typescript v5.2.2

### LINKED ISSUE #1237: Update ncids to use the typescript 5
# Update the NCIDS to use the latest version of typescript

```bash
[1/4] 🤔  Why do we have the module "typescript"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "typescript@4.5.4"
info Has been hoisted to "typescript"
info Reasons this module exists
   - "workspace-aggregator-39ee92e2-caba-4044-b724-c0855835efd1" depends on it
   - Hoisted from "_project_#@nciocpl#ncids-js#typescript"
   - Hoisted from "_project_#ncids-css-testing#typescript"
   - Hoisted from "_project_#ncids-js-testing#typescript"
   - Hoisted from "_project_#nci-design-system#typescript"
info Disk size without dependencies: "61.75MB"
info Disk size with unique dependencies: "61.75MB"
info Disk size with transitive dependencies: "61.75MB"
info Number of shared dependencies: 0
=> Found "lerna#typescript@4.9.5"
info This module exists because "_project_#lerna" depends on it.
info Disk size without dependencies: "64.06MB"
info Disk size with unique dependencies: "64.06MB"
info Disk size with transitive dependencies: "64.06MB"
info Number of shared dependencies: 0
✨  Done in 1.55s.
```

- [x] update ncids-js
- [x] update ncids-css-testing
- [x] update ncids-js-testing
- [x] remove from nci-design-system (doc site)

### PR DESCRIPTION
**Pull request by olitharp-nci**
- Remove dependency from documentation site

Closes #1237

### CODE REVIEW

**Comment by olitharp-nci** [comment]
```bash
[1/4] 🤔  Why do we have the module "typescript"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "typescript@5.2.2"
info Has been hoisted to "typescript"
info Reasons this module exists
   - "workspace-aggregator-e8381add-2084-49ef-8b10-e8cecf2f70ce" depends on it
   - Hoisted from "_project_#@nciocpl#ncids-js#typescript"
   - Hoisted from "_project_#ncids-css-testing#typescript"
   - Hoisted from "_project_#ncids-js-testing#typescript"
info Disk size without dependencies: "39.04MB"
info Disk size with unique dependencies: "39.04MB"
info Disk size with transitive dependencies: "39.04MB"
info Number of shared dependencies: 0
=> Found "lerna#typescript@4.9.5"
info This module exists because "_project_#lerna" depends on it.
info Disk size without dependencies: "64.06MB"
info Disk size with unique dependencies: "64.06MB"
info Disk size with transitive dependencies: "64.06MB"
info Number of shared dependencies: 0
✨  Done in 2.46s.
```

**Comment by olitharp-nci** [comment]
Will need to update eslint-config-vanilla-ts

https://github.com/NCIOCPL/cgov-standards-xt/blob/develop/packages/eslint-config-vanilla-ts/package.json#L19

```diff
- "@typescript-eslint/eslint-plugin": "^5.3.0",
- "@typescript-eslint/parser": "^5.3.0",
+ "@typescript-eslint/eslint-plugin": "^6.7.0",
+ "@typescript-eslint/parser": "^6.7.0",
```

File: packages/ncids-js/src/components/nci-autocomplete/__tests__/example-dom.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export const getExampleACDOMPlain = () => {
+export const getExampleACDOMPlain = (): HTMLElement => {
```
Comment (bryanpizzillo) [inline packages/ncids-js/src/components/nci-autocomplete/__tests__/example-dom.ts:1]: How did this ever squeak through with TS 4.5??

**Review by bryanpizzillo — APPROVED** [review]
Looks good!

**Review by arcepaul — APPROVED** [review]

**Pull request merged**